### PR TITLE
Add table aws_sagemaker_domain Closes #970

### LINKED ISSUE #970: Add table aws_sagemaker_domain
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-sagemaker-domain.html

### PR DESCRIPTION
**Pull request by karanpopat**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Outputs:

account_id = "533793682495"
aws_partition = "aws"
aws_region = "us-east-1"
resource_aka = "arn:aws:sagemaker:us-east-1:533793682495:domain/d-dtvlrpsmuyzj"
resource_id = "d-dtvlrpsmuyzj"
resource_name = "turbottest44961"

Running SQL query: test-get-query.sql
[
  {
    "account_id": "533793682495",
    "arn": "arn:aws:sagemaker:us-east-1:533793682495:domain/d-dtvlrpsmuyzj",
    "name": "turbottest44961",
    "partition": "aws",
    "region": "us-east-1"
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "akas": [
      "arn:aws:sagemaker:us-east-1:533793682495:domain/d-dtvlrpsmuyzj"
    ],
    "name": "turbottest44961"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "arn": "arn:aws:sagemaker:us-east-1:533793682495:domain/d-dtvlrpsmuyzj",
    "name": "turbottest44961",
    "partition": "aws",
    "region": "us-east-1"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "533793682495",
    "akas": [
      "arn:aws:sagemaker:us-east-1:533793682495:domain/d-dtvlrpsmuyzj"
    ],
    "region": "us-east-1",
    "title": "turbottest44961"
  }
]
✔ PASSED

POSTTEST: tests/aws_sagemaker_domain

TEARDOWN: tests/aws_sagemaker_domain

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```sql
> select
  name,
  arn,
  creation_time,
  status
from
  aws_sagemaker_domain;
+-----------------------+----------------------------------------------------------------+---------------------------+---------+
| name                  | arn                                                            | creation_time             | status  |
+-----------------------+----------------------------------------------------------------+---------------------------+---------+
| default-1651679351893 | arn:aws:sagemaker:us-east-1:533793682495:domain/d-izam8fpsou2h | 2022-05-04T21:19:13+05:30 | Pending |
+-----------------------+----------------------------------------------------------------+---------------------------+---------+

> select
  name,
  creation_time,
  home_efs_file_system_id,
  kms_key_id
from
  aws_sagemaker_domain
where 
  kms_key_id is null;
+-----------------------+---------------------------+-------------------------+------------+
| name                  | creation_time             | home_efs_file_system_id | kms_key_id |
+-----------------------+---------------------------+-------------------------+------------+
| default-1651679351893 | 2022-05-04T21:19:13+05:30 | fs-03c7a9b5279fe84a1    | <null>     |
+-----------------------+---------------------------+-------------------------+------------+
```
</details>

### CODE REVIEW

File: aws/table_aws_sagemaker_domain.go [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+			},
+			{
+				Name:        "name",
+				Description: "The domain name.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("DomainName"),
+			},
+			{
+				Name:        "arn",
+				Description: "The domain's Amazon Resource Name (ARN).",
```
Comment (ParthaI) [inline aws/table_aws_sagemaker_domain.go:43]:
```suggestion
				Description: "The Amazon Resource Name (ARN) of the domain.",
```

File: aws/table_aws_sagemaker_domain.go [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+//// HYDRATE FUNCTIONS
+
+func getAwsSageMakerDomain(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	var id string
+	if h.Item != nil {
+		id = sageMakerDomainId(h.Item)
+	} else {
+		id = d.KeyColumnQuals["id"].GetStringValue()
+	}
+
```
Comment (ParthaI) [inline aws/table_aws_sagemaker_domain.go:220]: Can we add an empty check for `id`?

File: docs/tables/aws_sagemaker_domain.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  name,
+  creation_time,
+  home_efs_file_system_id,
+  kms_key_id
+from
+  aws_sagemaker_domain
+where 
+  kms_key_id is null;
+```
+### List publicly accessible sagemaker domains
```
Comment (ParthaI) [inline docs/tables/aws_sagemaker_domain.md:32]:
```suggestion

### List publicly accessible sagemaker domains
```

File: docs/tables/aws_sagemaker_domain.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  name,
+  arn,
+  creation_time,
+  app_network_access_type
+from
+  aws_sagemaker_domain
+where 
+  app_network_access_type = 'PublicInternetOnly';
+  ```
+
```
Comment (ParthaI) [inline docs/tables/aws_sagemaker_domain.md:46]:
```suggestion
```

File: aws/table_aws_sagemaker_domain.go [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+			{
+				Name:        "app_network_access_type",
+				Description: "Specifies the VPC used for non-EFS traffic.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getAwsSageMakerDomain,
+			},
+			{
+				Name:        "app_security_group_management",
+				Description: "The entity that creates and manages the required security groups for inter-app communication in VPCOnly mode.",
```
Comment (misraved) [inline aws/table_aws_sagemaker_domain.go:61]:
```suggestion
				Description: "The entity that creates and manages the required security groups for inter-app communication in VPC Only mode.",
```
Should the description be changed as per the above suggestion?

File: aws/table_aws_sagemaker_domain.go [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+			},
+			{
+				Name:        "failure_reason",
+				Description: "The domain's failure reason.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getAwsSageMakerDomain,
+			},
+			{
+				Name:        "home_efs_file_system_id",
+				Description: "The ID of the Amazon Elastic File System (EFS) managed by this Domain.",
```
Comment (misraved) [inline aws/table_aws_sagemaker_domain.go:79]:
```suggestion
				Description: "The ID of the Amazon Elastic File System (EFS) managed by this domain.",
```

File: aws/table_aws_sagemaker_domain.go [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+			},
+			{
+				Name:        "default_user_settings",
+				Description: "Settings which are applied to UserProfiles in this domain if settings are not explicitly specified in a given UserProfile.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsSageMakerDomain,
+			},
+			{
+				Name:        "domain_settings",
+				Description: "A collection of Domain settings.",
```
Comment (misraved) [inline aws/table_aws_sagemaker_domain.go:131]:
```suggestion
				Description: "A collection of domain settings.",
```

File: aws/table_aws_sagemaker_domain.go [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsSageMakerDomain,
+			},
+			{
+				Name:        "subnet_ids",
+				Description: "The VPC subnets that Studio uses for communication.",
```
Comment (misraved) [inline aws/table_aws_sagemaker_domain.go:137]:
```suggestion
				Description: "The VPC subnets that studio uses for communication.",
```

**Review by ParthaI — APPROVED** [review]
LGTM

File: docs/tables/aws_sagemaker_domain.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+select
+  name,
+  arn,
+  creation_time,
+  status
+from
+  aws_sagemaker_domain;
+```
+
+### List sagemaker domains where efs volume is unencrypted
```
Comment (misraved) [inline docs/tables/aws_sagemaker_domain.md:19]:
```suggestion
### List sagemaker domains where EFS volume is unencrypted
```

**Review by misraved — COMMENTED** [review]
Make changes wherever required

File: docs/tables/aws_sagemaker_domain.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
+# Table: aws_sagemaker_domain
+
+An Amazon SageMaker Domain consists of an associated Amazon Elastic File System (Amazon EFS) volume; a list of authorized users; and a variety of security, application, policy, and Amazon Virtual Private Cloud (Amazon VPC) configurations.
```
Comment (misraved) [inline docs/tables/aws_sagemaker_domain.md:3]:
```suggestion
Amazon SageMaker Domain consists of an associated Amazon Elastic File System (Amazon EFS) volume; a list of authorized users; and a variety of security, application, policy, and Amazon Virtual Private Cloud (Amazon VPC) configurations.
```

**Review by misraved — APPROVED** [review]

**Pull request merged**